main.qml

```diff
@@ -1249,16 +1254,76 @@ ApplicationWindow {
+            }
+            onHistoryClicked: {
+                middlePanel.state = "History";
+                if(isMobile) hideMenu();
+                isReceived = false;
+                middlePanel.unlockedBalanceText = leftPanel.unlockedBalanceText =  isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.unlockedBalance);
+                middlePanel.balanceText = leftPanel.balanceText = isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.balance);
+            }
+            onAddressBookClicked: {
+                middlePanel.state = "AddressBook";
+                if(isMobile) hideMenu();
+                isReceived = false;
+                middlePanel.unlockedBalanceText = leftPanel.unlockedBalanceText =  isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.unlockedBalance);
+                middlePanel.balanceText = leftPanel.balanceText = isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.balance);
+            }
+            onMiningClicked: {
+                middlePanel.state = "Mining";
+                if(isMobile) hideMenu();
+                isReceived = false;
+                middlePanel.unlockedBalanceText = leftPanel.unlockedBalanceText =  isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.unlockedBalance);
+                middlePanel.balanceText = leftPanel.balanceText = isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.balance);
+            }
+            onSignClicked: {
+                middlePanel.state = "Sign";
+                if(isMobile) hideMenu();
+                isReceived = false;
+                middlePanel.unlockedBalanceText = leftPanel.unlockedBalanceText =  isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.unlockedBalance);
+                middlePanel.balanceText = leftPanel.balanceText = isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.balance);
+            }
+            onSettingsClicked: {
+                middlePanel.state = "Settings";
+                if(isMobile) hideMenu();
+                isReceived = false;
+                middlePanel.unlockedBalanceText = leftPanel.unlockedBalanceText =  isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.unlockedBalance);
+                middlePanel.balanceText = leftPanel.balanceText = isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.balance);
+            }
+            onKeysClicked: {
+                settingsPasswordDialog.open();
+                if(isMobile) hideMenu();
+                isReceived = false;
+                middlePanel.unlockedBalanceText = leftPanel.unlockedBalanceText =  isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.unlockedBalance);
+                middlePanel.balanceText = leftPanel.balanceText = isReceived ? "HIDDEN" : walletManager.displayAmount(currentWallet.balance);
```

Comment:
Lot's of code being repeated here. Could you factor out the common parts into an updateBalance() function that could be used by the onClicked-events and `onWalletUpdate()`?